.github/workflows/bazel.yml

```diff
@@ -0,0 +1,118 @@
+name: Bazel
+
+on: [pull_request, push]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  build-windows:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { name: "Windows (native)",             os: windows-2022, action: "test",  config: "--noenable_bzlmod --config=windows",     }
+          - { name: "Windows (arm)",                os: windows-2022, action: "build", config: "--noenable_bzlmod --config=windows_arm", }
+          - { name: "Windows (roborio)",            os: windows-2022, action: "build", config: "--noenable_bzlmod --config=roborio",     }
+
+    name: "Build ${{ matrix.name }}"
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+        with: { fetch-depth: 0 }
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'zulu'
+          java-version: 17
+          architecture: x64
+
+      - name: Build
+        run: bazel --output_user_root=C:\\bazelroot ${{ matrix.action }} -k ...   --config=ci ${{ matrix.config }} --verbose_failures
+        shell: bash
+
+  build-mac:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { name: "Mac (native)",            os: macos-14,   action: "test",  config: "--noenable_bzlmod --config=osx -c opt",      }
+          - { name: "Mac (roborio)",           os: macos-14,   action: "build", config: "--noenable_bzlmod --config=roborio -c opt",  }
+
+    name: "${{ matrix.name }}"
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v4
+        with: { fetch-depth: 0 }
+
+      - name: Build
+        run: bazel ${{ matrix.action }} -k ...   --config=ci ${{ matrix.config }} --nojava_header_compilation --verbose_failures
+        shell: bash
+
+  build-linux:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - { name: "Linux (native)",  os: ubuntu-22.04,   action: "test",  config: "--noenable_bzlmod --config=linux -c opt",    }
```

Comment:
Instead of requiring specifying `--config=<host_platform>` to target the host, let's use [`--enable_platform_specific_config`](https://bazel.build/run/bazelrc#enable_platform_specific_config).

Reply:
I tried that once but didn't figure out how to make it work with cross compilation. i.e. if you are trying to cross compile for roborio on windows, it will add both the windows flags and the roborio flags and not work.

I just pop the config in my user.bazelrc and switch if I want to build athena stuff. In a world where this gets filtered down to teams, the vscode plugin could autodetect which config to run based on context.

Reply:
Ah, hmm. I guess the alternative is to wrap the `cc_library` rule with one that `select`s the compiler flags.

Reply:
That gets messy if you have third party libraries because you'd have to patch their things to use the same macro you are using for your code if you want everything to be built with the same compiler flags.

Reply:
If you want, we have all the tool chains sandboxed for Linux with Bazel.